cla-backend-go/users/repository.go

```diff
@@ -277,6 +286,114 @@ func (repo repository) GetUserByUserName(userName string) (*models.User, error)
 	return nil, nil
 }
 
+func (repo repository) SearchUsers(searchField string, searchTerm string, fullMatch bool) (*models.Users, error) {
+	// Sorry, no results if empty company name
+	if strings.TrimSpace(searchTerm) == "" {
+		return &models.Users{
+			Users:          []models.User{},
+			LastKeyScanned: "",
+			ResultCount:    0,
+			SearchTerm:     searchTerm,
+			TotalCount:     0,
+		}, nil
+	}
+
+	queryStartTime := time.Now()
+
+	tableName := fmt.Sprintf("cla-%s-users", repo.stage)
+
+	// These are the columns we want returned
+	projection := buildUserProjection()
+
+	builder := expression.NewBuilder().WithProjection(projection)
+	// This is the filter we want to match
+	if fullMatch {
+		filter := expression.Name("user_name").Equal(expression.Value(searchTerm))
```

Comment:
Should this be `expression.Name(searchField).Equal(expression.Value(searchTerm))`?